include/sys/nvpair.h

```diff
@@ -181,29 +181,29 @@ _SYS_NVPAIR_H int nvlist_add_uint64(nvlist_t *, const char *, uint64_t);
 _SYS_NVPAIR_H int nvlist_add_string(nvlist_t *, const char *, const char *);
 _SYS_NVPAIR_H int nvlist_add_nvlist(nvlist_t *, const char *, nvlist_t *);
```

Comment:
Would you like to make the same change for `nvlist_add_nvlist()` while you're at it?
This might also apply to `nvlist_remove_nvpair()`. 

What about the `nvlist_t*` parameter to `nvlist_lookup_*()`?

(Same for the equivalent `fnvlist_*()` routines.)

Might apply to some of the `nvpair_*()` routines as well.

Reply:
I think you can't put const in the `nvlist_t *` parameter to the lookup functions for any types that we return by reference. The problem with that is that it would allow you to get a non-const pointer into a const nvlist by using the lookup functions. We can put const for all the ones that we return by value, though.

Reply:
So for these routines: 
```
int nvlist_lookup_uint8_array(nvlist_t *nvl, const char *name, uint8_t **a, uint_t *n)
int nvlist_lookup_nvlist(nvlist_t *nvl, const char *name, nvlist_t **val)
```
Is the thinking that the returned array (in `*a`)  or nvlist (`*val`) should be modifiable by the caller,  assuming they have exclusive access to the `nvl`?  I wasn't aware of that use case but I guess it would work, so supporting it in the API makes sense.

It would be hard to correctly modify the return value of `nvlist_lookup_string()`, since the string could be shortened (I think...) but not made longer.  Similarly, `nvlist_lookup_string_array()`, if you change one of the string pointers in the array (`char*`), I'm not sure how the memory ownership would work there.

Reply:
I tried to go through an add this everywhere I could, though some of it was complicated by the fact that `nvlist_{next,prev}_nvpair` modify the nvlist by modifying the `nvp_curr`